server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1022,15 +1024,15 @@ private void asyncCleanupUnlinkedShardLevelBlobs(
             listener.onResponse(null);
             return;
         }
-        threadPool.executor(ThreadPool.Names.SNAPSHOT).execute(ActionRunnable.wrap(listener, l -> {
-            try {
+        snapshotDeleteTaskRunner.enqueueTask(ActionListener.wrap(releasable -> {
```

Comment:
nit/suggestion (sorry Github won't let use its suggestion feature here):

```diff
diff --git a/server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java b/server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java
index dfdc09b696f..436f49a923e 100644
--- a/server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java
+++ b/server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java
@@ -1024,15 +1024,15 @@ public abstract class BlobStoreRepository extends AbstractLifecycleComponent imp
             listener.onResponse(null);
             return;
         }
-        snapshotDeleteTaskRunner.enqueueTask(ActionListener.wrap(releasable -> {
+        snapshotDeleteTaskRunner.enqueueTask(listener.map(releasable -> {
             try (releasable) {
                 deleteFromContainer(blobContainer(), filesToDelete);
-                listener.onResponse(null);
+                return null;
             } catch (Exception e) {
                 logger.warn(() -> format("%s Failed to delete some blobs during snapshot delete", snapshotIds), e);
                 throw e;
             }
-        }, listener::onFailure));
+        }));
     }

     // updates the shard state metadata for shards of a snapshot that is to be deleted. Also computes the files to be cleaned up.
```

NB slightly different semantics, this means we release `releasable` (potentially starting another delete task) before completing the grouped listener. I think that's ok tho.

Reply:
Thanks for the suggestion. All done in https://github.com/elastic/elasticsearch/pull/93593/commits/164b24815b6eae1c9b6950749857afdbc7b24169.